### DISCUSSION TITLE
Option to delay test execution in watch mode

### DISCUSSION THREAD

**Discussion by novemberborn** [avajs/ava#2702] [Ideas]
[Prompted](https://github.com/avajs/ava/issues/2061#issuecomment-786494003) by @jakobrosenberg, I'd like to explore options to delay test execution in watch mode. Specifically for use cases where tests are run against build output, and the build process takes a while to settle down.

@jakobrosenberg you mentioned Rollup's `buildDelay`, but also suggested a `diskIdle` option. Is that something you've seen elsewhere? How would you characterize the differences?

**Comment by jakobrosenberg** [comment]
> How would you characterize the differences?

I think I misphrased "better option". I really meant a better name (compared to `delay`, which can be very ambiguous). `diskIdle` on the other hand reminds me of `window.requestIdleCallback`, which I find self explanatory.

**Reply by novemberborn** [reply]
By "idle" you mean "this much time has passed since we last observed a file change"?

**Reply by jakobrosenberg** [reply]
Correct. It's a slight misnomer. 😳

**Reply by novemberborn** [reply]
No worries. See https://github.com/avajs/ava/issues/2715 for the issue.
